File: consul-examples/src/main/java/io/vertx/examples/consul/ConsulClientVerticle.java
package io.vertx.examples.consul;

import io.vertx.core.AbstractVerticle;
import io.vertx.core.Launcher;
import io.vertx.ext.consul.ConsulClient;

/**
 * @author <a href="mailto:ruslan.sennov@gmail.com">Ruslan Sennov</a>
 */
public class ConsulClientVerticle extends AbstractVerticle {

  /**
   * Convenience method so you can run it in your IDE
   */
  public static void main(String[] args) {
    Launcher.main(new String[]{"run", ConsulClientVerticle.class.getName()});
  }


  @Override
  public void start() {
    ConsulClient consulClient = ConsulClient.create(vertx);
    consulClient.putValue("key11", "value11")
      .compose(v -> {
        System.out.println("KV pair saved");
        return consulClient.getValue("key11");
      }).onComplete(ar -> {
      if (ar.succeeded()) {
        System.out.println("KV pair retrieved");
        System.out.println(ar.result().getValue());
      } else {
        ar.cause().printStackTrace();
      }
    });
  }
}

File: consul-examples/src/main/kotlin/io/vertx/examples/consul/ConsulClientVerticle.kt
package io.vertx.examples.consul

import io.vertx.ext.consul.ConsulClient

class ConsulClientVerticle : io.vertx.core.AbstractVerticle()  {
  override fun start() {
    var consulClient = ConsulClient.create(vertx)
    consulClient.putValue("key11", "value11").compose<Any>({ v ->
      println("KV pair saved")
      return consulClient.getValue("key11")
    }).onComplete({ ar ->
      if (ar.succeeded()) {
        println("KV pair retrieved")
        println(ar.result().value)
      } else {
        ar.cause().printStackTrace()
      }
    })
  }
}
